README.md

```diff
@@ -1,61 +1,143 @@
+
+    require 'rototiller'
+
+    desc "task dependencies work. this one also uses an environment variable"
+    rototiller_task :parent_task do |task|
+      # most method initializers take either a hash, or block syntax (see next task)
+      task.add_env({:name     => 'RANDOM_VAR', :default => 'default value'})
+      task.add_command({:name => "echo 'i am testing everything with $RANDOM_VAR = #{ENV['RANDOM_VAR']}'"})
+    end
+
+    desc "override command-name with environment variable"
+    rototiller_task :test => :parent_task do |task|
+      # block syntax here. We give up some lines for more readability
+      task.add_command do |cmd|
+        cmd.name         = 'test'
+        cmd.override_env = 'ECHO_EXECUTABLE'
+      end
+      task.add_flag({:name => '-f', :default => 'Rakefile'})
+    end
+
+    desc "override flag values with environment variables"
+    rototiller_task :test_flag_env do |task|
+      task.add_command do |cmd|
+        cmd.name = 'test'
+      end
+      task.add_flag do |flag|
+        flag.name         = '-f'
+        flag.default      = 'Rakefile'
+        flag.override_env = 'FLAG_VALUE'
+      end
+    end
+
+    desc "override command argument values with environment variables"
+    rototiller_task :test_arg_env do |task|
+      task.add_command do |cmd|
+        cmd.name                  = 'ls'
+        cmd.argument              = 'Rakefile'
+        cmd.argument_override_env = 'FILENAME'
+      end
+    end
+
+## Use
+(with the above sample Rakefile)
+
+    $) rake -T
+    rake parent_task  # some parent task
+    rake test         # test all the things
+
+    $) rake -D
+    rake parent_task
+        task dependencies work. this one also uses an environment variable
+    rake test
+        override command-name with environment variable
+
+    # added environment variable defaults are set, implicitly, if not found
+    #   this way, their value can be used in the task
+    $) rake test
+    INFO: The environment variable: 'RANDOM_VAR' was found with value: 'default value':
+    i am testing everything with $RANDOM_VAR = default value
+    The CLI flag -f will be used with value Rakefile.
+
+    $) rake parent_task RANDOM_VAR=redrum
+    INFO: The environment variable: 'RANDOM_VAR' was found with value: 'redrum':
+    i am testing everything with $RANDOM_VAR = redrum
+
+    $) rake test ECHO_EXECUTABLE='ls' --verbose
+    INFO: The environment variable: 'RANDOM_VAR' was found with value: 'default value':
+    echo 'i am testing everything with $RANDOM_VAR = default value'
+    i am testing everything with $RANDOM_VAR = default value
+    The CLI flag -f will be used with value Rakefile.
+
+    ls -f Rakefile
+    Rakefile
+
+    $) rake test_flag_env
+    The CLI flag -f will be used with value Rakefile.
+    $) echo $?
+    0
+
+    $) rake test_flag_env --verbose
+    The CLI flag -f will be used with value Rakefile.
+
+    test -f Rakefile
+
+    $) rake test_flag_env --verbose FLAG_VALUE='README.md'
```

Comment:
no messaging on where the value came from (what env, default or not). ticketed separately